fix/population-cap.lua

```diff
@@ -1,27 +1,30 @@
-local ui = df.global.plotinfo
-local ui_stats = ui.tasks
-local civ = df.historical_entity.find(ui.civ_id)
+local plotinfo = df.global.plotinfo
+local tasks = plotinfo.tasks
+local knowledge = tasks.knowledge
 
-if not civ then
+local civ = df.historical_entity.find(plotinfo.civ_id)
+
+if not dfhack.isMapLoaded() or not dfhack.world.isFortressMode() or not civ then
     qerror('No active fortress.')
 end
 
 local civ_stats = civ.activity_stats
 
-if not civ_stats then
-    civ.activity_stats = {
-        new = true,
-        created_weapons = { resize = #ui_stats.created_weapons },
-        discovered_creature_foods = { resize = #ui_stats.discovered_creature_foods },
-        discovered_creatures = { resize = #ui_stats.discovered_creatures },
-        discovered_plant_foods = { resize = #ui_stats.discovered_plant_foods },
-        discovered_plants = { resize = #ui_stats.discovered_plants },
-    }
-    civ_stats = civ.activity_stats
+local function upsize(civ_vec, tasks_vec)
+    local tasks_vec_size = #tasks_vec
+    if #civ_vec < tasks_vec_size then
+        civ_vec:resize(tasks_vec_size)
+    end
 end
 
+upsize(civ_stats.created_weapons, tasks.created_weapons)
```

Comment:
I rewrote the logic here because the original didn't seem to make sense -- `civ.activity_stats` is not a pointer